src/classes/QueryDateLiteral.cls

```diff
@@ -4,185 +4,194 @@
 *************************************************************************************************/
 public without sharing class QueryDateLiteral {
 
-    private final String LAST_N = 'LAST_N_{0}: {1}';
-    private final String NEXT_N = 'NEXT_N_{0}: {1}';
-
-    private final String DAYS            = 'DAYS';
-    private final String WEEKS           = 'WEEKS';
-    private final String MONTHS          = 'MONTHS';
-    private final String QUARTERS        = 'QUARTERS';
-    private final String FISCAL_QUARTERS = 'FISCAL_QUARTERS';
-    private final String YEARS           = 'YEARS';
-    private final String FISCAL_YEARS    = 'FISCAL_YEARS';
+    private final static String LAST_N = 'LAST_N_{0}: {1}';
+    private final static String NEXT_N = 'NEXT_N_{0}: {1}';
+    private final static String N_DAYS_AGO = 'N_DAYS_AGO: {0}';
+
+    private final static String DAYS            = 'DAYS';
+    private final static String WEEKS           = 'WEEKS';
+    private final static String MONTHS          = 'MONTHS';
+    private final static String QUARTERS        = 'QUARTERS';
+    private final static String FISCAL_QUARTERS = 'FISCAL_QUARTERS';
+    private final static String YEARS           = 'YEARS';
+    private final static String FISCAL_YEARS    = 'FISCAL_YEARS';
 
     private String value;
 
     public String getValue() {
         return this.value;
     }
 
+    private QueryDateLiteral(String value) {
+        this.value = value;
+    }
+
     //Actual constant literals
-    public QueryDateLiteral YESTERDAY {
-        get {return this.setValue('YESTERDAY');}
+    public static QueryDateLiteral YESTERDAY {
```

Comment:
haha I just realized QueryArgumentFormatter still needs to be converted, but that can be done at a later time